app/utils.py

```diff
@@ -34,71 +31,65 @@ def cosine_similarity(peer_v, query_v):
     return num / (sqrt(den_a) * sqrt(den_b))
 
 
-def loadEntropies():
-    global root_dir
+def load_entropies(entropies_file='demo/ukwac.entropy.txt'):
     entropies_dict = {}
-    entropies_file = os.path.join(
-            root_dir, "demo",
-            "ukwac.entropy.txt")
-    f = open(entropies_file, "r")
-    for l in f:
-        l = l.rstrip('\n')
-        fields = l.split('\t')
-        w = fields[0].lower()
-        # Must have this cos lower() can match two instances of the same word in the list
-        if w.isalpha() and w not in entropies_dict:
-            entropies_dict[w] = float(fields[1])
-    f.close()
+    with open(entropies_file, "r") as entropies:
+        for line in entropies:
+            word, score = line.split('\t')
+            word = word.lower()
+            # Must have this cos lower() can match two instances of the same word in the list
+            if word.isalpha() and word not in entropies_dict:
+                entropies_dict[word] = float(score)
+
     return entropies_dict
 
 
-def mkQueryDist(query, entropies):
+def query_distribution(query, entropies):
     """ Make distribution for query """
     words = query.rstrip('\n').split()
 
     # Only retain arguments which are in the distributional semantic space
     vecs_to_add = []
-    for w in words:
-        word = OpenVectors.query.filter(OpenVectors.word == w).first()
+    for word in words:
+        word = OpenVectors.query.filter(OpenVectors.word == word).first()
         if word:
             vecs_to_add.append(word)
         else:
-            w = w[0].upper() + w[1:]  # Did user carelessly forget to capitalise a proper noun?
-            word = OpenVectors.query.filter(OpenVectors.word == w).first()
+            word = word[0].upper() + word[1:]  # Did user carelessly forget to capitalise a proper noun?
+            word = OpenVectors.query.filter(OpenVectors.word == word).first()
             if word:
                 vecs_to_add.append(word)
 
     vbase = array([])
     # Add vectors together
-    if len(vecs_to_add) > 0:
+    if vecs_to_add:
         # Take first word in vecs_to_add to start addition
         vbase = array([float(i) for i in vecs_to_add[0].vector.split(',')])
-        for item in range(1, len(vecs_to_add)):
-            w = vecs_to_add[item]
-            if w in entropies and math.log(entropies[w] + 1) > 0:
-                weight = float(1) / float(math.log(entropies[w] + 1))
-                vbase = vbase + weight * array([float(i) for i in vecs_to_add[item].vector.split(',')])
+        for vec in vecs_to_add[1:]:
+            print(vec.word in entropies)
+            if vec.word in entropies and math.log(entropies[vec.word] + 1) > 0:
```

Comment:
checking `if vec  in entropies` was always `False` because vec was `OpenVector` model instance and keys in entropies are strings, thus else branch was always executed.